cfme/configure/settings.py

```diff
@@ -1,173 +1,228 @@
-        view = navigate_to(self, 'Copy')
+    def copy(self, description=None, cancel=False):
+        """
+        This method performs the copy of the provided time profile object.
+        Args:
+            description (str) : It's the descriptive name of the new copied time_profile.
+            cancel (bool) : This variable performs cancel operation while copy.
+
+        return: It returns the object of the copied time_profile.
+        """
 
-        if name is not None:
-            new_timeprofile = Timeprofile(description=name, scope=self.scope)
-            changed = view.timeprofile_form.fill({
-                'description': name,
+        view = navigate_to(self, 'Copy')
+        if description is not None:
+            new_time_profile = self.parent.instantiate(description=description, scope=self.scope)
+            changed = view.form.fill({
+                'description': description,
                 'scope': self.scope,
             })
         else:
-            new_timeprofile = Timeprofile(description="{} copy".format(self.description),
-                                          scope=self.scope)
-            changed = view.timeprofile_form.fill({
+            new_time_profile = self.parent.instantiate(
+                description="{} copy".format(self.description),
+                scope=self.scope
+            )
+            changed = view.form.fill({
                 'description': "{} copy".format(self.description),
                 'scope': self.scope,
             })
 
-        if changed:
-            view.timeprofile_form.save_button.click()
-        return new_timeprofile
+        if not cancel and changed:
+            view.form.add.click()
+            view.flash.assert_no_error()
+        return new_time_profile
+
+
+@attr.s
+class TimeProfileCollection(BaseCollection):
+
+    ENTITY = TimeProfile
+
+    def create(self, description=None, scope=None, days=True, hours=True,
+               timezone=None, cancel=False):
+        """
+        Args:
+            description (str): It's the descriptive name of the time_profile.
+            scope: It's the option 'All User' or 'Current User' from dropdown.
+            days (bool): It's the option to switch on or switch off the days Bootstrap switch.
+            hours (bool): It's the option to switch on or switch off the hours Bootstrap switch.
+            timezone: It's the required Time Zone for the time_profile.
+            cancel (bool) : It's a flag used to cancel or not the create operation.
+
+        return: It returns the object of the newly created time_profile object.
+        """
+
+        time_profile = self.instantiate(description, scope, days, hours, timezone)
+
+        view = navigate_to(self, 'Add')
+        view.form.fill({
+            'description': description,
+            'scope': scope,
+            'days': days,
+            'hours': hours,
+            'timezone': timezone,
+        })
+        if not cancel:
```

Comment:
**REQUIRED**
Here again, you return a new instance of `TimeProfile` even if Cancel button was clicked.